### PR TITLE
Upgrade issues

### PR DESCRIPTION
**Pull request by amithkanand**
This PR fixes issues that I found during upgrade of LAB cluster from 2.2 to 2.3.4.

### CODE REVIEW

File: cookbooks/bcpc-hadoop/attributes/mapred.rb [hunk 1]
```diff
@@ -15,6 +15,10 @@
   hdp_path =
     File.join('/usr/hdp',
               node[:bcpc][:hadoop][:distribution][:active_release])
+
+  hdp_apps_path =
+    File.join('/hdp/apps',
```
Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/attributes/mapred.rb]: Should this have the HDFS path prepended?

File: cookbooks/bcpc-hadoop/attributes/yarn.rb [hunk 2]
```diff
@@ -90,21 +90,29 @@
 
   site_xml['yarn.nodemanager.remote-app-log-dir'] = '/var/log/hadoop-yarn/apps'
   
-  avail_memory = node['bcpc']['hadoop']['yarn']['nodemanager']['avail_memory']
-  site_xml['yarn.nodemanager.resource.memory-mb'] =
-    avail_memory['size'] or
-    [1024, (node['memory']['total'].to_i * avail_memory['ratio']/1024).floor].max
+  yarn_max_memory = lambda do
```
Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/attributes/yarn.rb]: Huh? Why a function here. What are you avoiding referencing before it is set?

File: cookbooks/bcpc-hadoop/attributes/mapred.rb (hunk 1 above)
Comment (amithkanand) [inline cookbooks/bcpc-hadoop/attributes/mapred.rb]: We never had HDFS parth prepended in the past and I kept it the same way.

File: cookbooks/bcpc-hadoop/attributes/yarn.rb (hunk 2 above)
Comment (http-418) [inline cookbooks/bcpc-hadoop/attributes/yarn.rb]: I don't know what the underlying issue is.  This was a "try it and see."  I haven't had time to revisit the problem.

**Pull request merged**